felix.go

```diff
@@ -321,7 +321,11 @@ configRetry:
 	// Initialise the glue logic that connects the calculation graph to/from the dataplane driver.
 	log.Info("Connect to the dataplane driver.")
 	failureReportChan := make(chan string)
-	dpConnector := newConnector(configParams, datastore, dpDriver, failureReportChan)
+	var configUpdChan chan map[string]string
+	if configParams.UsageReportingEnabled {
+		configUpdChan = make(chan map[string]string, 1)
+	}
+	dpConnector := newConnector(configParams, configUpdChan, datastore, dpDriver, failureReportChan)
```

Comment:
I first read this as "creating a connector object with an initial set of config, and with a channel so that we can also pass config changes to the connector".  Whereas in fact I think the channel is for the connector to pass config changes to *its* downstream(s).

Given that, would it be clearer for the channel creation to move inside newConnector?  It might also need moving the whole of the usagerep setup inside newConnector - but perhaps that would make sense too?

Alternatively, perhaps just a comment here to explain.

Reply:
Added a comment and made the naming clearer.